### PR TITLE
Connect Metamask

### PR DESCRIPTION
**Pull request by habibitcoin**
Ready to go!

Allows you to connect to either nosft.xyz, generative.xyz, ordswap.io or alby. Metamask options only shown if metamask is installed. Sends working

https://github.com/dannydeezy/nosft/issues/29

To-dos:
- [x] Create wallet selection modal
- [x] Verify sends/receives work properly
- [x] Change generative.xyz -> nosft.xyz
- [x] Fix linting/cleanup
- [x] Allow switching back to nosft.xyz
- [x] Styling

### CODE REVIEW

File: src/components/modals/send-modal/index.js [hunk 1]
```diff
@@ -39,12 +45,39 @@ const SendModal = ({ show, handleModal, utxo, onSale }) => {
...
+
     async function sendUtxo() {
         const inputAddressInfo = getAddressInfo(nostrPublicKey);
+        const domain = SessionStorage.get(SessionsStorageKeys.DOMAIN);
         const psbt = new bitcoin.Psbt({
             network: TESTNET ? bitcoin.networks.testnet : bitcoin.networks.bitcoin,
         });
-        const publicKey = Buffer.from(await window.nostr.getPublicKey(), "hex");
+        const { ethereum } = window;
+        const ethAddress = ethereum.selectedAddress;
```
Comment (andredevjs) [inline src/components/modals/send-modal/index.js:60]: if ethereum is undefined, would this break the code?

File: src/layouts/header.js [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
 import WalletContext from "@context/wallet-context";
 import headerData from "../data/general/header.json";
 import menuData from "../data/general/menu.json";
 
 const Header = React.forwardRef(({ className, onConnectHandler, onDisconnectHandler }, ref) => {
     const { offcanvas, offcanvasHandler } = useOffcanvas();
-    const { nostrPublicKey, nostrAddress } = useContext(WalletContext);
+    const { nostrPublicKey, nostrAddress, ethProvider } = useContext(WalletContext);
+
+    console.log('header loaded');
```
Comment (andredevjs) [inline src/layouts/header.js:23]: remove debug statements

File: src/pages/index.js [hunk 3]
```diff
@@ -40,15 +42,27 @@ const App = () => {
         setNostrAddress(address);
     }, [nostrPublicKey]);
 
+    useEffect(() => {
+        if (typeof window == undefined) return;
```
Comment (andredevjs) [inline src/pages/index.js:46]: debug statements

File: src/services/session-storage.js [hunk 4]
```diff
@@ -2,6 +2,7 @@ const SessionsStorageKeys = {
     NOSTR_PUBLIC_KEY: "NOSTR_PUBLIC_KEY",
     INSCRIPTIONS_ON_SALE: "INSCRIPTIONS_ON_SALE",
     INSCRIPTIONS_OWNED: "INSCRIPTIONS_OWNED",
+    DOMAIN: "DOMAIN",
```
Comment (andredevjs) [inline src/services/session-storage.js:5]: what does domain means?

File: .eslintignore [hunk 5]
```diff
@@ -34,3 +34,6 @@ yarn-error.log\*
 
 # vercel
 .vercel
+
+# temp
+/src
```
Comment (andredevjs) [inline .eslintignore:39]: why are you ignoring eslint?

**Comment by andredevjs** [comment]
this looks very promising! great job
@habibitcoin no worries about UI.
Let me know when code is ready and I will fix it

**Comment by andredevjs** [comment]
@habibitcoin added fixes here: https://github.com/habibitcoin/nosft/pull/1

File: src/components/modals/send-modal/index.js (hunk 1 above)
Comment (habibitcoin) [reply src/components/modals/send-modal/index.js]: Nope, still works ok

**Comment by andredevjs** [comment]
ready to rock

**Pull request merged**